pkg/internal/testabilities/testservices/fixture_bhs.go

```diff
@@ -0,0 +1,113 @@
+package testservices
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/defs"
+	"github.com/go-softwarelab/common/pkg/to"
+	"github.com/jarcoal/httpmock"
+)
+
+type longestChainTipResponse struct {
+	Height        uint   `json:"height"`
+	Hash          string `json:"hash"`
+	Version       uint32 `json:"version"`
+	MerkleRoot    string `json:"merkleRoot"`
+	Timestamp     uint64 `json:"creationTimestamp"`
+	Bits          uint64 `json:"bits"`
+	Nonce         uint32 `json:"nonce"`
+	PreviousBlock string `json:"prevBlockHash"`
+}
+
+type LongestChainTipOptions func(*longestChainTipResponse)
+
+func WithLongestChainTipHeight(h uint) LongestChainTipOptions {
+	return func(l *longestChainTipResponse) {
+		l.Height = h
+	}
+}
+
+type BHSFixture interface {
+	IsUpAndRunning() BHSFixture
+	WillBeUnreachable() error
+	WillRespondWithInternalFailure()
+	WillRespondWithEmptyLongestTipBlockHeader()
+	OnLongestTipBlockHeaderResponseWith(opts ...LongestChainTipOptions)
+}
+
+type bhsFixture struct {
+	testing.TB
+	transport       *httpmock.MockTransport
+	longestChainTip *longestChainTipResponse
+}
+
+func (b bhsFixture) WillRespondWithEmptyLongestTipBlockHeader() {
+	b.transport.RegisterResponder(
+		http.MethodGet, defs.BHSTestURL+"/chain/tip/longest",
+		httpmock.NewJsonResponderOrPanic(http.StatusOK, longestChainTipResponse{}),
+	)
+}
+
+func (b *bhsFixture) OnLongestTipBlockHeaderResponseWith(opts ...LongestChainTipOptions) {
+	for _, o := range opts {
+		o(b.longestChainTip)
+	}
+}
+
+var bhsAnyEndpointRegexFixture = regexp.MustCompile(fmt.Sprintf(`^http:\/\/%s(?:\/.*)?$`, defs.BHSTestURL[7:]))
```

Comment:
if i am correct you could simply do
```suggestion
var bhsAnyEndpointRegexFixture = regexp.MustCompile(`^http://(?:localhost:\d+)(?:/.*)?$`)
```
what is this magic number 7 ?